### ISSUE TITLE
Suggestion - For making a billing machine in Food business and Mechanic shop

### ISSUE THREAD

**Issue by Kabir7700** [InZidiuZ/op-framework-issue-tracker#2671]
### Assurance

I have read the `README.md`

### Summary

As we have all know that when we deliver food / do repair we have to take money from customer in seller/ mechanic either in account or cash.

=>But instead of this we can add a billing machine in food shop ( Burgershot , bean machine and Japanese restaurant) and Mechanic shops where we can directly deduct money from Customer account accounting their PayPal ( Server ID). 

=> And when we ask for pay pal customer will get a request for the transaction in their phone and once they accept the request then only transaction can be made ! And also ………. 

=> For ex. Customer buys a total of $1000 worth items !  60-70% can be transferred to The person who makes this bill and 30-40% money out of it gets stored in the Company! 

=> so here by owner doesn’t have to go after people who work with them for their weekly-commissions. 

=> Also it’s more realistic.

### Reason

This may enhance food business RP , Mechanic RP and also provide Easy to all the concerned people in billing and deducting the amount direct from customer.

**Comment by Niveet** [comment]
I mean, this would be a huge QOL improvement. Currently, it's a pain to manage all of it on discord and 3rd party softwares. I think, you guys should implement it command based tho, it should provide the functionality and should be easy to understand to new players.

**Comment by Liltears33** [comment]
@InZidiuZ I believe this would be a pretty big improvement when you have the time, the QOL for business owners is pretty rough because a lot of employees will lie about what they sell. Personally I've received no commission unless I'm working my own restaurant. Love you master like always. ~Tommy Turner

**Comment by Puucko4090** [comment]
=> For ex. Customer buys a total of $1000 worth items ! 60-70% can be transferred to The person who makes this bill and 30-40% money out of it gets stored in the Company! employees will love earning 30-40% less on on a product they payed to sell in the first place, this means they get to keep like 10-20% ppl will just stop work 100%in thease places instead of 80% cus the pay is bad as it is

**Comment by CloudTheWolf** [comment]
So my main question would be, how exactly would this work for tips?

Also this would not work with the current implementation for mechanic shops since they are all part of the "LS Customs" job

**Comment by Kabir7700** [comment]
> => For ex. Customer buys a total of $1000 worth items ! 60-70% can be transferred to The person who makes this bill and 30-40% money out of it gets stored in the Company! employees will love earning 30-40% less on on a product they payed to sell in the first place, this means they get to keep like 10-20% ppl will just stop work 100%in thease places instead of 80% cus the pay is bad as it is

I mean ………..
The point here is ……. Dev.s make an in game billing machine where billing can give majority of amount to seller and rest to the amount can be transferred to owner.
And as far is rate is concerned it can be set by Dev.s

**Comment by Kabir7700** [comment]
> stored

30-40 is an example. Limit can be set by the Dev.s ! 
It can be 20-25 as well. And also I’ve noticed restaurants selling 

![89D4EEF1-2FFB-4818-BF3A-787C71C0088F](https://user-images.githubusercontent.com/116685592/217321057-2c967629-10b4-4b00-a027-ae19f50ebbee.png)

We can add an option like this which also can be transferred to the seller’s account without any deduction to company ( 100% to seller’s account ) .

**Comment by Kabir7700** [comment]
> So my main question would be, how exactly would this work for tips?
> 
> Also this would not work with the current implementation for mechanic shops since they are all part of the "LS Customs" job

![A1B1FC23-5530-4F25-9F5B-E3884765CF5F](https://user-images.githubusercontent.com/116685592/217321764-a0ca100c-3967-47b5-a0a1-c744a9d898df.png)

Dev.s can add an option like this in the same machine. 100% amount of which will be directly transferred to seller’s account and 0% to company!

**Comment by Puucko4090** [comment]
there is one way i can see this working whitout beeing at he expense of the employees, that is if the owner get the cash the employee pays to get the product out of the store that they sell to custumores, because that cost is already there and 50% of that shoulde go to keep the store stock/material up so the employees got sumthing to sell

**Comment by Kabir7700** [comment]
> there is one way i can see this working whitout beeing at he expense of the employees, that is if the owner get the cash the employee pays to get the product out of the store that they sell to custumores, because that cost is already there and 50% of that shoulde go to keep the store stock/material up so the employees got sumthing to sell

Yes 
But what if an employee wants to make 5 burgers and 1 burger cost the company 70 dollar. And he makes 5 for himself and don’t sell it. So company will face loss of 350 and if I’ve 15 employees I’ll have to face loss of $5,500 dollar each day from my pocket! 🥲😅
Weekly 35,000. From which I can pay city hall / mayor, 3 times.

**Comment by CloudTheWolf** [comment]
yeah in the current process I'm pretty sure the employee buys the ingredients at wholesale price, then uses them and sells the burger at a profit.

Feels like it's trying to over complicate the process so that business owners make more.

**Comment by Puucko4090** [comment]
> > there is one way i can see this working whitout beeing at he expense of the employees, that is if the owner get the cash the employee pays to get the product out of the store that they sell to custumores, because that cost is already there and 50% of that shoulde go to keep the store stock/material up so the employees got sumthing to sell
> 
> Yes But what if an employee wants to make 5 burgers and 1 burger cost the company 70 dollar. And he makes 5 for himself and don’t sell it. So company will face loss of 350 and if I’ve 15 employees and everyone will have an urge to do so. Because who would not like to gett food on company’s expense. I’ll have to face loss of $5,500 dollar each day from my pocket! 🥲😅 Weekly 35,000. From which I can pay city hall / mayor, 3 times.

if an employee buys the ingrediants for a burger owner coulde get his money from that what happens to that burger is up to the employee if he sells it or eat it make monnie or fill belly simple easy

**Comment by Puucko4090** [comment]
and how woulde this appliy to mechanic shops ea mechanic buys there own materials shoulde owner get a peice of that seams weird and greedy

**Comment by Kabir7700** [comment]
> > > there is one way i can see this working whitout beeing at he expense of the employees, that is if the owner get the cash the employee pays to get the product out of the store that they sell to custumores, because that cost is already there and 50% of that shoulde go to keep the store stock/material up so the employees got sumthing to sell
> > 
> > 
> > Yes But what if an employee wants to make 5 burgers and 1 burger cost the company 70 dollar. And he makes 5 for himself and don’t sell it. So company will face loss of 350 and if I’ve 15 employees and everyone will have an urge to do so. Because who would not like to gett food on company’s expense. I’ll have to face loss of $5,500 dollar each day from my pocket! 🥲😅 Weekly 35,000. From which I can pay city hall / mayor, 3 times.
> 
> if an employee buys the ingrediants for a burger owner coulde get his money from that what happens to that burger is up to the employee if he sells it or eat it make monnie or fill belly simple easy

Ohh I got your point now ! I misunderstood it priorly. 
Hahaha 
But what you’re saying …….Sounds good too !

**Comment by Liltears33** [comment]
As a business owner my employees don't pay me anything they sell stuff and I do agree that they should be paid more for work. But they aren't responsible to buy the company. The owner is, so what I'm hearing is a company owner must work consistent hours to pay for the business because it's not a small price. Employees on our whitelisted companies also get a higher paycheck than non whitelisted jobs. I'm not saying a high percentage goes to the owner like previously listed but 5-10% going to the owner isn't a bad idea considering they did the work to get the business and also have the power to terminate based on the companies policies. I do believe they should make more than what they pay for it being an employee but why does a owner not get any money when an employee sells their exclusive merchandise? Is my only question

**Comment by CloudTheWolf** [comment]
> > and how woulde this appliy to mechanic shops ea mechanic buys there own materials shoulde owner get a peice of that seams weird and greedy
> 
> To be honest ! When I used to work as mechanic 3/4 months back. We used to pay weekly to owner. And we had a calculator as well. So I though we use the same thing like having a calculator and if calculation gets upto $2000 for a car ( super car ) then owner can get 400 and rest can be transferred to mechanic who repaired it. => but as you mentioned something about “LA “. I’m really not aware about that. So can’t comment over it anymore !

This just wouldn't work for a mechanic shop with how it currently works in the server as every mechanic shop is part of the "Mechanic=>Los Santos Customs" Job (Hence why you _technically_ can unlock door at any mechanic shop regardless of where you work.

**Comment by coalaura** [comment]
Most of this suggestion is not needed. The only part that can be looked into is making a system to bill someone a certain amount and them being able to pay said bill

**Comment by Kabir7700** [comment]
@coalaura what do you think about the company’s cut from the same bill ? Adding 10-15% company cut over that bill ?

**Comment by coalaura** [comment]
no, like i said thats not really needed. We don't need new separate company accounts for every single possible RP company. That can be done in RP if needed

**Comment by coalaura** [comment]
You can now do /bill [server id] [amount]
